src/components/card/DueDateSelector.vue

```diff
@@ -64,64 +83,81 @@ export default defineComponent({
 				stringify: this.stringify,
 				parse: this.parse,
 			},
-			shortcuts: [
-				{
-					text: t('deck', 'Today'),
-					onClick() {
-						const date = new Date()
-						date.setDate(date.getDate())
-						date.setHours(23)
-						date.setMinutes(59)
-						return date
-					},
-				},
-				{
-					text: t('deck', 'Tomorrow'),
-					onClick() {
-						const date = new Date()
-						date.setDate(date.getDate() + 1)
-						date.setHours(23)
-						date.setMinutes(59)
-						return date
-					},
-				},
-				{
-					text: t('deck', 'Next week'),
-					onClick() {
-						const date = new Date()
-						date.setDate(date.getDate() + 7)
-						date.setHours(23)
-						date.setMinutes(59)
-						return date
-					},
-				},
-				{
-					text: t('deck', 'Next month'),
-					onClick() {
-						const date = new Date()
-						date.setDate(date.getDate() + 30)
-						date.setHours(23)
-						date.setMinutes(59)
-						return date
-					},
-				},
-			],
 		}
 	},
 	computed: {
 		duedate: {
 			get() {
-				return this.card.duedate ? new Date(this.card.duedate) : null
+				return this.card?.duedate ? new Date(this.card.duedate) : null
 			},
 			set(val) {
-				this.$emit('change', val)
+				this.$emit('change', val ? new Date(val) : null)
 			},
 		},
+
+		reminderOptions() {
+			const currentDateTime = moment()
```

Comment:
We need to give proper attribution to this https://github.com/nextcloud/spreed/blob/24c1aa87c632b210583bfaf1eee16ecaa67f3ee4/src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue#L529

Might be nice to add this to the moment library wrapper